### PR TITLE
Remove opentelemetry during prod, in Rust

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
none, followup to https://github.com/grapl-security/grapl/pull/1817

### What changes does this PR make to Grapl? Why?
We were seeing in CI that grapl-core failed to deploy due to 
> OpenTelemetry trace error occurred. Exporter jaeger encountered the following error(s): thrift agent failed with not open

This gates the rust opentelemetry stuff behind "does the OTEL_EXPORTER_JAEGER_AGENT_HOST env var exist?"

I then standardized how we feed in the otel jaeger/zipkin stuff into services. It'll populate those env vars in local but not in prod, just like aws_env_vars_for_local. 

### How were these changes tested?
I overrode `observability_env_vars_for_local` locally so it'd look like prod; stuff started up!

### CODE REVIEW

**Comment by twunderlich-grapl** [comment]
I'll note that with Nomad 1.3.2, I should have a way forward for prod

File: nomad/grapl-core.nomad [hunk 1]
```diff
@@ -1131,12 +1163,11 @@ job "grapl-core" {
         PLUGIN_EXECUTION_IMAGE                          = var.container_images["generator-execution-sidecar"] # TODO: add support for analyzer too
         PLUGIN_REGISTRY_BUCKET_AWS_ACCOUNT_ID           = var.plugin_registry_bucket_aws_account_id
         PLUGIN_REGISTRY_BUCKET_NAME                     = var.plugin_registry_bucket_name
+        PLUGIN_EXECUTION_OBSERVABILITY_ENV_VARS         = var.observability_env_vars
```
Comment (twunderlich-grapl) [inline nomad/grapl-core.nomad:1166]: Why does this need a separate env var that just repeats the existing env var?

Comment (twunderlich-grapl) [reply nomad/grapl-core.nomad:1166]: Ah, this is passed into the plugins. Got it.

**Review by twunderlich-grapl — APPROVED** [review]

File: nomad/grapl-core.nomad (hunk 1 above)
Comment (wimax-grapl) [reply nomad/grapl-core.nomad:1166]: So, we pass through some vars from
`grapl-core.nomad` -> plugin-registry -> `hax_docker_plugin.nomad` -> plugin binaries

what we're doing to plugin-registry is:
- passing it env vars: `OTEL_BLAH_BLAH=1, ...` that plugin-registry uses at runtime
- passing it a template string: "OTEL_BLAH_BLAH=1\n..." that plugin-registry passes down to the plugin binaries

By passing in the template string, we basically **remove some complexity around the optional-ness of the env vars** (the reason we did the aws_env_vars_for_local stuff in the first place) + make hax_docker_plugin.nomad more-resemble grapl-core.nomad

**Pull request merged**